source/common/http/alternate_protocols_cache_impl.cc

```diff
@@ -3,14 +3,27 @@
 #include "source/common/common/logger.h"
 
 #include "quiche/spdy/core/spdy_alt_svc_wire_format.h"
+#include "re2/re2.h"
 
 namespace Envoy {
 namespace Http {
-namespace {
-std::string originToString(const AlternateProtocolsCache::Origin& origin) {
+
+std::string
+AlternateProtocolsCacheImpl::originToString(const AlternateProtocolsCache::Origin& origin) {
   return absl::StrCat(origin.scheme_, "://", origin.hostname_, ":", origin.port_);
 }
-} // namespace
+
+absl::optional<AlternateProtocolsCache::Origin>
+AlternateProtocolsCacheImpl::stringToOrigin(const std::string& str) {
+  re2::RE2 origin_regex("(.*)://(.*):(\\d+)");
```

Comment:
I don't love regular expression in "real" code (perf/memory), but I guess this isn't on the hot path, right? Alternatively we could serialize by joining with some delimiter and then use split? Or we could use google URL to do the parsing / serializing? *shrug*

Would it make sense to make this a member of the class so we only initialize it once? Or is that not important because it's not on the hot path?